query/src/main/scala/filodb/query/exec/rangefn/RangeFunction.scala

```diff
@@ -225,82 +226,128 @@ trait ChunkedLongRangeFunction extends TimeRangeFunction[TransientRow] {
+            useDownsampledColumns: Boolean = false,
             maxCol: Option[Int] = None,
             useChunked: Boolean): BaseRangeFunction =
-    generatorFor(func, columnType, config, funcParams, maxCol, useChunked)()
+    generatorFor(dataset, func, columnType, config, funcParams, useChunked)()
 
   /**
    * Given a function type and column type, returns a RangeFunctionGenerator
    */
-  def generatorFor(func: Option[RangeFunctionId],
+  def generatorFor(dataset: Dataset,
+                   func: Option[RangeFunctionId],
                    columnType: ColumnType,
                    config: QueryConfig,
                    funcParams: Seq[Any] = Nil,
-                   maxCol: Option[Int] = None,
-                   useChunked: Boolean = true): RangeFunctionGenerator =
+                   useChunked: Boolean = true): RangeFunctionGenerator = {
     if (useChunked) columnType match {
-      case ColumnType.DoubleColumn => doubleChunkedFunction(func, config, funcParams)
-      case ColumnType.LongColumn   => longChunkedFunction(func, funcParams)
-      case ColumnType.TimestampColumn => longChunkedFunction(func, funcParams)
-      case ColumnType.HistogramColumn => histChunkedFunction(func, funcParams, maxCol)
-      case other: ColumnType       => throw new IllegalArgumentException(s"Column type $other not supported")
+      case ColumnType.DoubleColumn => doubleChunkedFunction(dataset, func, config, funcParams)
+      case ColumnType.LongColumn => longChunkedFunction(dataset, func, funcParams)
+      case ColumnType.TimestampColumn => longChunkedFunction(dataset, func, funcParams)
+      case ColumnType.HistogramColumn => histChunkedFunction(dataset, func, funcParams)
+      case other: ColumnType => throw new IllegalArgumentException(s"Column type $other not supported")
     } else {
       iteratingFunction(func, funcParams)
     }
+  }
 
   /**
    * Returns a function to generate a ChunkedRangeFunction for Long columns
    */
-  def longChunkedFunction(func: Option[RangeFunctionId],
-                          funcParams: Seq[Any] = Nil): RangeFunctionGenerator = func match {
-    case None                 => () => new LastSampleChunkedFunctionL
-    case Some(CountOverTime)  => () => new CountOverTimeChunkedFunction()
-    case Some(SumOverTime)    => () => new SumOverTimeChunkedFunctionL
-    case Some(AvgOverTime)    => () => new AvgOverTimeChunkedFunctionL
-    case Some(MinOverTime)    => () => new MinOverTimeChunkedFunctionL
-    case Some(MaxOverTime)    => () => new MaxOverTimeChunkedFunctionL
-    case Some(StdDevOverTime) => () => new StdDevOverTimeChunkedFunctionL
-    case Some(StdVarOverTime) => () => new StdVarOverTimeChunkedFunctionL
-    case _                    => iteratingFunction(func, funcParams)
+  def longChunkedFunction(dataset: Dataset,
+                          func: Option[RangeFunctionId],
+                          funcParams: Seq[Any] = Nil): RangeFunctionGenerator = {
```

Comment:
I prefer to not change the indentation and keep func match on the previous line.

Reply:
Can we please keep the extra paren? Makes it easier to read and add code.